client/components/SingleProduct.js

```diff
@@ -20,8 +21,24 @@ class SingleProduct extends Component {
     await this.props.getReviews(productId)
   }
 
+  handleSubmit(event) {
+    event.preventDefault()
+    try {
+      console.log(this.props, 'all the props')
+      let orderItem = {
+        quantity: this.state.quantityToAdd,
+        productId: this.props.match.params.productId,
+        orderId: this.props.cart.id
```

Comment:
there is cart.id here, but if the user is a guest, is there an order id?  How does this affect the route.post('/cart') ?